### PR TITLE
tidb-engine-ext: add ut pipeline

### PR DESCRIPTION
**Pull request by purelind**
* add ut pipeline for tidb-engine-ext

### CODE REVIEW

File: pipelines/pingcap/tidb-engine-ext/latest/pod-ghpr_unit_test.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  securityContext:
+    fsGroup: 1000
+  containers:
+    - name: golang
+      image: "hub.pingcap.net/wangweizhen/tidb_image:go11920220921"
+      tty: true
+      resources:
+        requests:
+          memory: 8Gi
+          cpu: "2"
```
Comment (wuhuizuo) [inline pipelines/pingcap/tidb-engine-ext/latest/pod-ghpr_unit_test.yaml:13]: how about add resource limits?

File: pipelines/pingcap/tidb-engine-ext/latest/ghpr_unit_test.groovy [hunk 2]
```diff
@@ -0,0 +1,85 @@
+// REF: https://www.jenkins.io/doc/book/pipeline/syntax/#declarative-pipeline
+// Keep small than 400 lines: https://issues.jenkins.io/browse/JENKINS-37984
+// should triggerd for master and release-6.2.x branches
+final K8S_NAMESPACE = "jenkins-tiflash"
```
Comment (wuhuizuo) [inline pipelines/pingcap/tidb-engine-ext/latest/ghpr_unit_test.groovy:4]: please add a git ops dir in [this folder](https://github.com/PingCAP-QE/ee-ops/tree/main/apps/prod/jenkins/post) later.

**Review by wuhuizuo — APPROVED** [review]

File: jobs/pingcap/tidb-engine-ext/latest/ghpr_unit_test.groovy [hunk 3]
```diff
@@ -0,0 +1,84 @@
+// REF: https://<your-jenkins-server>/plugin/job-dsl/api-viewer/index.html
+// For trunk and latest release branches.
+pipelineJob('pingcap/tidb-engine-ext/ghpr_unit_test') {
+    logRotator {
+        daysToKeep(90)
```
Comment (wuhuizuo) [inline jobs/pingcap/tidb-engine-ext/latest/ghpr_unit_test.groovy:5]: 30 or 90?

**Review by wuhuizuo — APPROVED** [review]

File: pipelines/pingcap/tidb-engine-ext/latest/ghpr_unit_test.groovy (hunk 2 above)
Comment (purelind) [reply pipelines/pingcap/tidb-engine-ext/latest/ghpr_unit_test.groovy:4]: I will add it later

File: jobs/pingcap/tidb-engine-ext/latest/ghpr_unit_test.groovy (hunk 3 above)
Comment (purelind) [reply jobs/pingcap/tidb-engine-ext/latest/ghpr_unit_test.groovy:5]: Low trigger frequency, retaining three months is more reasonable

**Review by CalvinNeo — APPROVED** [review]
lgtm. thx for your help~

**Pull request merged**